query.go

```diff
@@ -66,6 +66,10 @@ func NewQuery(ctx *Context, array *Array) (*Query, error) {
 
 // Free tiledb_query_t that was allocated on heap in c
 func (q *Query) Free() {
+	q.bufferMutex.Lock()
+	defer q.bufferMutex.Unlock()
+	q.buffers = nil
+	q.resultBufferElements = nil
 	if q.tiledbQuery != nil {
 		C.tiledb_query_free(&q.tiledbQuery)
```

Comment:
Should we set `q.tiledbQuery` to nil here, to prevent a double-free? Same for `func (b *Buffer) Free()`

Reply:
`q.tiledbQuery` is set to nil/nullptr by the `C.tiledb_query_free` function.

Reply:
Ah, thanks!